01 OperatorOverloading/OperatorOverloading.Model/Money.cs

```diff
@@ -77,20 +88,63 @@ public Money(string inputAmount)
             {
                 throw new InvalidCurrencyException(Messages.NullValue);
             }
+
             if (obj1.Currency.Equals(obj2.Currency, StringComparison.CurrentCultureIgnoreCase))
             {
                 double totalAmount = obj1.Amount + obj2.Amount;
                 return new Money(totalAmount, obj1.Currency);
             }
+
+            if (obj1.Currency.Equals("usd", StringComparison.CurrentCultureIgnoreCase))
+            {
+                var temp = obj1.Convert(obj2.Currency);
+                return new Money(temp.Amount + obj2.Amount, obj2.Currency);
+            }
             else
             {
-                throw new InvalidCurrencyException(Messages.MismatchedCurrency);
+                double exchangeRate = ExchangeRate(obj1.Currency, obj2.Currency);
+                return new Money(obj1.Amount * exchangeRate + obj2.Amount, obj2.Currency);
             }
         }
 
         public override string ToString()
         {
-            return Amount + " " + Currency;
+            return Amount + " " + Currency.ToUpper();
+        }
+
+        /// <summary>
+        /// Accepts a string of currency to be converted to.
+        /// </summary>
+        /// <param name="convertTo"></param>
+        /// <returns></returns>
+        public Money Convert(string convertTo)
+        {
+            if (Currency.Equals("usd", StringComparison.CurrentCultureIgnoreCase))
+            {
+                double result = FetchResult(Currency, convertTo);
+                return new Money(result * Amount, convertTo);
+            }
+            else 
+            {
+                double exchangeRate = ExchangeRate(Currency, convertTo);
+                return new Money(exchangeRate * Amount, convertTo);
+            }
+        }
+
+        private static double ExchangeRate(string convertFrom, string convertTo)
+        {
+            return FetchResult("USD", convertTo) / FetchResult("USD", convertFrom);
```

Comment:
Too compact code, hard to debug. Will break if DBL returns 0 from `FetchResult("USD", convertFrom)`